### PR TITLE
[fix] Allow subscription cancellation.

### PR DESCRIPTION
**Pull request by primarilysnark**
This PR allows for subscription cancellation during the onMessage loop. Previous versions threw constraint violations during the `onMessage` `forEach`.

### CODE REVIEW

File: lib/socket.dart [hunk 1]
```diff
@@ -46,7 +46,7 @@ class PhoenixSocket<T> {
   }) async =>
       PhoenixSocket<T>._(
         encoding: encoding,
-        ws: await WebSocket.connect('$address/websocket'),
```
Comment (Schoonology) [inline lib/socket.dart]: This seems like an unrelated—and seemingly undesired—change.

**Review by Schoonology — CHANGES_REQUESTED** [review]

File: lib/socket.dart (hunk 1 above)
Comment (primarilysnark) [reply lib/socket.dart]: Unrelated, yes. Undesired, no. It prevents the use of query parameters on the websocket address. That said, I'll bring this up in another PR.

Comment (Schoonology) [reply lib/socket.dart]: Ohhhhh, right. We discussed this earlier. Forgot.

We should probably still take care of the `/websocket` part, if only because it's a non-obvious assumption of Phoenix. 🤷‍♂️

**Review by Schoonology — APPROVED** [review]

File: lib/socket.dart (hunk 1 above)
Comment (amiel) [reply lib/socket.dart]: The config endpoint now provides this full url, and it includes the `/websocket` part.
